rir/src/compiler/pir/instruction.h

```diff
@@ -1022,25 +1045,27 @@ class FLIE(Extract1_1D, 3, Effects::Any()) {
     }
 };
 
-class FLIE(Extract2_1D, 3, Effects::Any()) {
+class FLIE(Extract2_1D, 3, Effects::Any() & ~Effects(Effect::Visibility)) {
```

Comment:
you seem to have mixed in stuff from abandoned commits from less-effects, stopping review here...